src/test/app/TransferRate_test.cpp

```diff
@@ -0,0 +1,64 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <test/jtx.h>
+#include <ripple/beast/unit_test.h>
+#include <algorithm>
+
+namespace ripple {
+namespace test {
+
+// Make sure transfer rate 
+class TransferRate_test : public beast::unit_test::suite
+{
+public:
+    void
+    test (std::uint32_t tr, std::uint32_t aliceBal)
+    {
+        using namespace jtx;
+        Env env(*this);
+        auto const gw = Account("gateway");
+        auto const USD = gw["USD"];
+        env.fund(XRP(10000), gw, "alice", "bob");
+        env.trust(USD(2), "alice", "bob");
+        env.close();
+		auto jt = noop("alice");
+        jt[sfTransferRate.fieldName] = tr;
+        env(jt);
+        env.close();
+        env(pay(gw, "alice", USD(2)));
+        env(pay("alice", "bob", USD(1)));
+        env.require(balance("alice", USD(aliceBal)));
+        env.require(balance("bob", USD(1)));
+    }
+
+    void
+    run() override
+    {
+        test(1000000000, 1);
+        test(2000000000, 0);
```

Comment:
I agree with the preceding comments.  I'd be curious to see a bunch of test cases.  You might start with:
- 0
- 1
- `QUALITY_ONE` - 1
- `QUALITY_ONE`
- `QUALITY_ONE`*2
- `QUALITY_ONE`*2 + 1
- `std::numeric_limits<std::uint32_t>::max()`.

Then try those four cases with and without the amendment enabled.

Then try applying the transfer rate to `alice` (as you're currently doing), but also to `gw` alone and also to both `alice` and `gw`.  I suspect applying it to `gw` is more typical of what we'd expect.  But it's also good to see what happens if a non-issuer (like alice) sets a transfer rate.

FWIW, there is a (tiny) pre-existing transfer rate test in AccountSet_test.cpp.  Might it make sense for your test(s) to go in there?  I'm not sure that putting your tests in a new file is a net win...

I just looked a bit more at AccountSet_test.cpp.  A few of the tests I've suggested are covered there.  That may argue for keeping all of the transfer rate tests in the same file so it's a bit easier to see when there's redundancy.